lib/iiif/v3/abstract_resource.rb

```diff
@@ -5,22 +5,37 @@ module V3
     class AbstractResource
       include IIIF::HashBehaviours
 
-      # Every subclass should override the following methods defining key value types, see Subclasses for how
+      # properties used by content resources only
+      CONTENT_RESOURCE_PROPERTIES = %w{ format height width duration }
+
+      # used by Collection, AnnotationCollection
+      PAGING_PROPERTIES = %w{ first last next prev total start_index }
```

Comment:
these will be used by subclasses to make it a lot easier to specify prohibited_keys.  The spec is full of language like "Other resource types must not have a xxx" and these are two groups of properties set that are basically used together, or at least in the same way.